File: src/Stronghold.Application/Features/Staff/StaffResponse.cs
namespace Stronghold.Application.Features.Staff;

public class StaffResponse
{
    public int Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Bio { get; set; }
    public string? ProfileImageUrl { get; set; }
    public string StaffType { get; set; } = string.Empty;
    public bool IsActive { get; set; }
}

File: stronghold_desktop/lib/features/staff/models/staff_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'staff_response.freezed.dart';
part 'staff_response.g.dart';

@freezed
abstract class StaffResponse with _$StaffResponse {
  const factory StaffResponse({
    required int id,
    required String firstName,
    required String lastName,
    required String email,
    String? phone,
    String? bio,
    String? profileImageUrl,
    required String staffType,
    required bool isActive,
  }) = _StaffResponse;

  factory StaffResponse.fromJson(Map<String, dynamic> json) =>
      _$StaffResponseFromJson(json);
}

@freezed
abstract class PagedStaffResponse with _$PagedStaffResponse {
  const factory PagedStaffResponse({
    required List<StaffResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedStaffResponse;

  factory PagedStaffResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedStaffResponseFromJson(json);
}
